### ISSUE TITLE
Numbers in select field are converted to string

### ISSUE THREAD

**Issue by thopedam** [measuredco/puck#713]
Using the following configuration the value is being converted to a string

```javascript
{
  fields: {
    numberField: {
      type: "select", // "radio"
      label: "A Select-Field with numbers",
      options: [
        { value: 1, label: "1" },
        { value: 2, label: "2" },
        { value: 3, label: "3" },
        { value: 4, label: "4" },
        { value: 5, label: "5" },
        { value: 6, label: "6" },
        { value: 7, label: "7" },
        { value: 8, label: "8" },
        { value: 9, label: "9" },
        { value: 10, label: "10" },
        { value: 11, label: "11" },
        { value: 12, label: "12" },
      ],
    },
    resolveData: (data, {changed}) {
      console.log(JSON.stringify(data, null, 2));
    }
  }
}
```

This returns:

```
{
...
  "props": {
    "numberField": "6"
  }
...
}
```
This happens because internally the HTML input values are string.

Maybe something like this is helpful for SelectField and RadioField? I am maybe missing something, so this is a bug/discussion and not a MR.  

```
const isNumberField =
    typeof value === "number" ||
    field.options.filter((option) => typeof option.value === "number")?.length >
      0;
```

**Comment by chrisvxd** [comment]
We already support booleans in radio/select, so I'm happy to consider this a bug and extend that implementation to numbers! Marking as ready.

**Comment by chrisvxd** [comment]
Closed by #721
